src/hooks/useBestTrade.ts

```diff
@@ -32,16 +36,28 @@ export function useBestTrade(
     200
   )
 
+  const wethContract = useWETHContract()
+  const isAWrapTransaction = useMemo(() => {
+    if (!wethContract || !chainId || !amountSpecified || !debouncedOtherCurrency) return undefined
```

Comment:
This should only return a boolean, and then you shouldn't need to cast `shouldGetTrade` to a boolean.

```suggestion
    if (!wethContract || !chainId || !amountSpecified || !debouncedOtherCurrency) return false
```